packages/shorebird_cli/lib/src/cache.dart

```diff
@@ -162,17 +162,64 @@ abstract class CachedArtifact {
       return false;
     }
 
-    if (checksum == null) {
+    final expectedChecksum = sha256Checksum;
+    if (expectedChecksum == null) {
       logger.detail(
         '''No checksum provided for $fileName, skipping file corruption validation''',
       );
       return true;
     }
 
-    return checksumChecker.checkFile(file, checksum!);
+    return checksumChecker.checkFile(
+      file,
+      checksum: expectedChecksum,
+      algorithm: ChecksumAlgorithm.sha256,
+    );
   }
 
   Future<void> update() async {
+    try {
+      final response = await _makeDownloadRequest();
+      final tempFile = await _downloadFile(response);
+
+      final isZip =
+          response.headers[HttpHeaders.contentTypeHeader] == 'application/zip';
+
+      _verifyChecksum(file: tempFile, responseHeaders: response.headers);
+
+      // Create the directory containing the artifact if it does not already
+      // exist. Failing to do this will cause [renameSync] to throw an
+      // exception.
+      final parentDirectory = file.parent..createSync(recursive: true);
+      if (isZip) {
+        final unzipDirectory = Directory(
+          p.join(p.dirname(tempFile.path), fileName),
+        );
+        await artifactManager.extractZip(
+          zipFile: tempFile,
+          outputDirectory: unzipDirectory,
+        );
+        await copyPath(unzipDirectory.path, parentDirectory.path);
```

Comment:
Is this the final install?  If this gets interrupted here will we end up with a corrupted cache?

Reply:
Yes to both. I believe our discussion on the call was that `renameSync` was not atomic. This seemed easier to read, although it's probably slower and thus more prone to being interrupted, so maybe `renameSync` is the way to go, even if it doesn't 100% guarantee that it can't be cancelled in flight. wdyt?

Reply:
Saw your comment below, disregard this